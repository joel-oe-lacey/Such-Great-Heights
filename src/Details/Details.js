import React from 'react';
import './Details.scss';
import Carousel from '../Carousel/Carousel'
import PropTypes from 'prop-types';

// import { withRouter } from 'react-router';



const Details = ({ listings, match, addFavorite, removeFavorite })=> {
    const listing = listings.find(listing => listing.id === parseInt(match.params.listing_id))
    //below is undefined handling, if no listing is available, use empty data from listings
    const { id, name, details: { address, beds, baths, cost_per_night, features }} = listing ? listing : listings[0];

    const uniqueFeatures = features.map(feature => {
        return <li>{feature}</li>
    })

    return (
        <div className='details-container'>
            <section className='details-card'>
                <Carousel className='carousel' id={id} name={name} imgTags={['a','b','c']} />
                <button className='details-fav' onClick={() => addFavorite(listing)}>Add Favorite</button>
                <button className='details-unfav' onClick={() => removeFavorite(listing)}>Remove Favorite</button>
                <ul className='details-info'>
                    <h3 className='details-list'>{name}</h3>
                    <h4 className='details-list'>{address}</h4>
                    <li className='details-list'>{`Number of bedrooms: ${beds}`}</li>
                    <li className='details-list'>{`Number of bathrooms: ${baths}`}</li>
                    <li className='details-list'>{`Cost per night: $${cost_per_night}`}</li>
                    <h4 className='unique-features'> Unique Features
                        <ul className='unique-features'>
                            {uniqueFeatures}
                        </ul>
                    </h4>
                </ul>
            </section>
        </div>
    )
}

export default Details;

Details.propTypes = {
  listings: PropTypes.array,
  match: PropTypes.object,
  addFavorite: PropTypes.func,
  removeFavorite: PropTypes.func
};
