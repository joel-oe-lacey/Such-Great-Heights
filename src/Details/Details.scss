.details-container {
  height: 100%;
  background-color: #89B0AE;

}

.details-card {
  height: 70%;
  display: grid;
  grid-template-columns: 25% 50% 25%;
  grid-template-rows: 40% 15% 15% 15% 15%;
  grid-template-areas:
    '. carousel .'
    '. . .'
    'favorite details .'
    'unfavorite details .'
    '. unique .';
  justify-content: center;
  align-content: center;
  font-family: 'Work Sans', sans-serif;

}

.carousel {
  margin-top: 10px;
  grid-area: carousel;
  height: 100%;
  width: 100%;
  justify-content: center;
}

.details-fav {
  grid-area: favorite;
}

.details-unfav {
  grid-area: unfavorite;
}

.details-fav,
.details-unfav {
  width: 50%;
  justify-self: center;
  align-self: center;
  height: 40%;
  border-radius: 4px;
  font-family: 'Work Sans', sans-serif;
  font-weight: 700;
  font-size: 1em;
  background-color: #FFD6BA;
  cursor: pointer;
}

.details-info {
  grid-area: details;
  justify-self: center;
  align-self: flex-start;
}

.details-list {
  margin: 10px;

}

.unique-features {
  grid-area: unique;
}
