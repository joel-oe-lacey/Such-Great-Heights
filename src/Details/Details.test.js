import React from 'react';
import { shallow } from 'enzyme';
import Details from './Details/Details';

describe('Details', () => {
    it('should equal snapshot ', () => {
        const listings = [{"id":3,"area_id":590,"name":"Hip RiNo Party Spot","details":{"address":"2250 Lawrence St, 80205","neighborhood_id":5124122,"superhost":true,"seller_source":"91jss1","beds":3,"baths":2.5,"cost_per_night":420,"features":["hot tub","espresso machine"]}}, 
        { "id": 2, "area_id": 751, "name": "The Other House", "details": { "address": "1800 Safe Way, 80202", "neighborhood_id": 5124122, "superhost": true, "seller_source": "91jss1", "beds": 3, "baths": 2.5, "cost_per_night": 420, "features": ["hot tub", "espresso machine"] } 
        }]; 
        const matchMock = {params: {listing_id: 2}}
        const wrapper = shallow(<Details listings={listings} match={matchMock} />);
        expect(wrapper).toMatchSnapshot();
        //how to test that it will display a different listing if listing match is different?
    });

})