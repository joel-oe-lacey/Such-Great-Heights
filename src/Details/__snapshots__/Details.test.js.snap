// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details should equal snapshot  1`] = `
<div
  className="details-container"
>
  <section
    className="details-card"
  >
    <Carousel
      className="carousel"
      id={2}
      imgTags={
        Array [
          "a",
          "b",
          "c",
        ]
      }
      name="The Other House"
    />
    <button
      className="details-fav"
      onClick={[Function]}
    >
      Add Favorite
    </button>
    <button
      className="details-unfav"
      onClick={[Function]}
    >
      Remove Favorite
    </button>
    <ul
      className="details-info"
    >
      <h3
        className="details-list"
      >
        The Other House
      </h3>
      <h4
        className="details-list"
      >
        1800 Safe Way, 80202
      </h4>
      <li
        className="details-list"
      >
        Number of bedrooms: 3
      </li>
      <li
        className="details-list"
      >
        Number of bathrooms: 2.5
      </li>
      <li
        className="details-list"
      >
        Cost per night: $420
      </li>
      <h4
        className="unique-features"
      >
         Unique Features
        <ul
          className="unique-features"
        >
          <li>
            hot tub
          </li>
          <li>
            espresso machine
          </li>
        </ul>
      </h4>
    </ul>
  </section>
</div>
`;
